src/Idris/DocString.idr

```diff
@@ -242,7 +243,7 @@ getDocsForName fc n
                TCon _ _ _ _ _ _ cons _
                    => do let tot = [showTotal n (totality d)]
                          cdocs <- traverse (getDConDoc <=< toFullNames) cons
-                         let cdoc = case concat cdocs of
+                         let cdoc = case cdocs of
```

Comment:
This used to collapse a list of empties to an empty list which we are not doing anymore?

Reply:
I noticed that `getDConDoc` was always either returning an empty list or a singleton, so I changed that to just return a Doc (and `Empty` in case no documentation could be found), so no `concat` should be necessary anymore.
I'm not sure if `getDConDoc` can ever return `Empty`, since we are passing in a fully qualified Name for a data constructor which is known to exist, so the lookup should never fail.

Reply:
For a type with no data constructors (tested with `Void`) this works as expected and does not show the "Constructor(s):" header

Reply:
What if we have a single constructor? Do we get everything on a single line?

Reply:
Ok, I found several Names for which `getDConDoc` returns empty (e.g. `Control.ANSI.MkDString`). I'll investigate an update this PR

Reply:
Two insights:
a) For single-constructor types, everything is shown on one line
b) For records, no constructor is shown, ever (but this was the case before this change, so it seems I didn't break anything that was working before).

It seems it's not possible to add a docstring to the constructor of a record, is that correct?

Reply:
> It seems it's not possible to add a docstring to the constructor of a record, is that correct?

Looking at the parser, that looks correct.
This is definitely something we could change.

Reply:
I just added documentation output for record constructors, and added test cases for single-constructor types and records